src/toil_scripts/gatk_germline/germline.py

```diff
@@ -1,688 +1,357 @@
-    assert os.path.exists(file_path)
-    return job.fileStore.writeGlobalFile(file_path)
-
-
-def return_input_paths(job, work_dir, ids, *filenames):
-    """
-    Given one or more filenames, reads the file from the fileStore and
-    copies it (or makes a hard link) into the working directory.
-
-    :param job: Job instance
-    :param work_dir: working directory, str
-    :param ids: dictionary of shared file ids
-    :param filenames: remaining arguments are filenames
-
-    :returns: list of paths to files
+def gatk_germline_pipeline(job, uuid, url, config, rg_line=None):
     """
-    paths = OrderedDict()
-    for filename in filenames:
-        if not os.path.exists(os.path.join(work_dir, filename)):
-            file_path = job.fileStore.readGlobalFile(ids[filename], os.path.join(work_dir, filename))
-        else:
-            file_path = os.path.join(work_dir, filename)
-        paths[filename] = file_path
+    Runs GATK Germline Pipeline on a single sample. Writes gvcf and vqsr.vcf to output directory.
 
-    return paths.values()
-
-
-def write_to_filestore(job, work_dir, ids, *filenames):
+    :param job: Toil Job instance
+    :param uuid str: Unique identifier for the sample
+    :param url str: URL to sample bam or fq file
+    :param config dict: Configuration options for pipeline
     """
-    Given one or more file names in working directory, writes
-    files to filestore and stores the filestore promise in a dictionary
+    config = deepcopy(config)
+    config['uuid'] = uuid
+    config['rg_line'] = rg_line
 
-    :param job: Job instance
-    :param work_dir: working directory
-    :param ids: shared file promises, dict
-    :param filenames: remaining arguments are keys for ids
+    cores = multiprocessing.cpu_count()
+    # Determine how much RAM is available
+    if config ['xmx'] is None:
+        config['xmx'] = '{}G'.format(os.sysconf('SC_PAGE_SIZE') * os.sysconf('SC_PHYS_PAGES') /
+                                     (1024 ** 3))
 
-    :returns: shared ids dictionary
-    """
-    for filename in filenames:
-        ids[filename] = job.fileStore.writeGlobalFile(os.path.join(work_dir, filename))
-    return ids
 
+    # Determine extension on input file
+    base, ext1 = os.path.splitext(url)
+    _, ext2 = os.path.splitext(base)
 
-def read_from_filestore_hc(job, work_dir, ids, *filenames):
-    """
-    Reads file from fileStore and writes it to working directory.
-
-    :param job: Job instance
-    :param work_dir: working directory
-    :param ids: shared file promises, dict
-    :param filenames: remaining arguments are filenames
-    """
-    for filename in filenames:
-        if not os.path.exists(os.path.join(work_dir, filename)):
-            job.fileStore.readGlobalFile(ids[filename], os.path.join(work_dir, filename))
+    # Produce or download BAM file
+    if config['run_bwa'] and rg_line and {ext1, ext2} & {'.fq', '.fastq'}:
```

Comment:
We need to dedupe between here and adam_gatk_pipeline. IMO, adam_gatk_pipeline should be the entry point for the germline pipeline.